### DISCUSSION TITLE
Retrieving and Setting main form values from Modal form

### DISCUSSION THREAD

**Discussion by tangjh** [filamentphp/filament#1594] [Help] [answered]
I'm using Admin Resource - Form and trying to open up a modal form to display a calculated value (retrieve and calculated from main form values) into TextInput in the modal form.  On clicking Ok button in modal form, the value of TextInput in modal form need to be send back to a disabled TextInput in the main form.

Is this possible?  Where should I:
1) retrieve the main form values for the calculation
2) how do I populate back the calculated value to main form?

```
class CreateItem extends CreateRecord
{
    protected static string $resource = ItemResource::class;

    protected function getActions(): array
    {
        return [
            ButtonAction::make('RetrieveVehicleData')
                ->action(function (array $data): void {
                         // what's the command to populate MyValue back to main form TextInput?
                })
                ->form([
                    Forms\Components\TextInput::make('MyValue') // how to retrieve and calculate values from main form?
                        ->label('MyValue')
                        ->required(),
                ]),
        ];
    }
}
```

**Comment by danharrin** [comment, marked as answer]
You can set the value back using `$this->data['field_name'] = $data['MyValue']`

To fill it when the action is first loaded, you could use `->default($this->data['field_name'])` on the TextInput?

**Comment by tangjh** [comment]
Ah .. Thanks!  Miss that out.
